### DISCUSSION TITLE
Allow handlebars templating for the extends field

### DISCUSSION THREAD

**Discussion by lstoeferle** [renovatebot/renovate#27952] [Suggest an Idea]
### Tell us more.

Hi everyone,
would it be possible to allow handlebars templating for the `extends` field like in the following example:

Self-Hosted configuration
```js
module.exports = {
  customEnvVariables: {
    GITLAB_REF: process.env.CI_COMMIT_REF_NAME,
  },
}
```

Repository configuration

```
{
  extends: ["local>foo/bar#{{ env.GITLAB_REF }}"]
}
```

This could be quite nice in case you are hosting the custom presets within the same repository. We for example run the Renovate config-validator as a CI job on branch level. Unfortunately we need to add the `#my-branch-name` ref to all the local included presets, else Renovate would just simply fetch the files from the default branch.

I'm happy to work on a PR in case you are fine with adding such feature (if even possible).

Kind regards,
Lukas

**Comment by rarkins** [comment]
I think this should be ok if you can find an elegant way to implement it

**Comment by lstoeferle** [comment]
Referring https://github.com/renovatebot/renovate/pull/27955#issuecomment-2003480562:

### Status Quo

- We have a single GitaLab repository `my-org/renovate` containing the self-hosted config (`config.js`) for Renovate
- Renovate runs on a scheduled basis via GitLab CI.
- Additionally this repository contains a default preset (`default.json`), which is also used in the `config.js` via [local extends](https://docs.renovatebot.com/config-presets/#self-hosted-git)

Simplified repository tree

```
- presets/
  - group/
  - ...
- .gitlab-ci.yml
- config.js
- default.json
```

Simplified `config.js`

```js
// config.js
module.exports = {
  extends: ["local>my-org/renovate"]
  ...
}
```

### The Problem

For each merge request we run the `renovate-config-validator`, since we want to make sure that the changes will not break the Renovate CI job (e.g. due to wrong repository configurations). To be able to test this we need to put a reference (in this case the branch name) to the included presets, since Renovate is using the default branch as reference by default. This is quite annoying, since we need to remove the reference before merging the changes, which then leads to a wrong pipeline run (fetching presets from the default branch).

Example:

```
// config.js
module.exports = {
  extends: [
    "local>my-org/renovate#my-branch-name"
  ]
}
```

### The Solution?

Allow templating like in the example on [top](https://github.com/renovatebot/renovate/discussions/27952#discussion-6377721). So we can forward the GitLab CI variable `CI_COMMIT_REF_NAME ` to Renovate, which the can be used as a reference.

**Comment by lstoeferle** [comment]
Feature included in the following release https://github.com/renovatebot/renovate/releases/tag/37.363.0
